src/dnaio/exceptions.py

```diff
@@ -43,3 +43,28 @@ class FastaFormatError(FileFormatError):
     """
 
     format = "FASTA"
+
+
+class IsNotAscii(ValueError):
```

Comment:
I think the subclassing of the error does not add much unless there is a benefit to catching only errors of this class. 
In this case I think the ValueError is more informative. There is something wrong with a supplied value. The ASCII error part is already stated in the message.
Wouldn't a function that throws a ValueError with the same error message be a better option here?

Reply:
Yeah, I’ve pondered various options and maybe haven’t settled on the best one.

In the end, what I want to achieve is that `FastaReader.__iter__` throws a `FastaFormatError` whenever there’s a problem in the FASTA file. At the moment, it’s inconsistent because some parsing problems lead to `FastaFormatError`, but some to a `ValueError` (they bubble up from `SequenceRecord.__init__`). This is why I made the change below where I wrap `yield SequenceRecord...` in a `try: ... except`.

Since `IsNotAscii` is caught and immediately converted to a `FastaFormatError`, it may not make much sense to have an extra exception class just for this. I’ll change this.

Reply:
Another idea was to just throw the `UnicodeEncodeError` (it’s a subclass of ValueError), but that loses the information *which* of the values wasn’t ASCII (name/sequence/qualities).